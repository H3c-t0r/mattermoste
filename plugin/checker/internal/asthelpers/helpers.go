// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

package asthelpers

import (
	"go/ast"
	"go/types"

	"github.com/pkg/errors"
	"golang.org/x/tools/go/packages"
)

func GetPackage(pkgPath string) (*packages.Package, error) {
	cfg := &packages.Config{
		Mode: packages.NeedName | packages.NeedTypes | packages.NeedSyntax,
	}
	pkgs, err := packages.Load(cfg, pkgPath)
	if err != nil {
		return nil, err
	}

	if len(pkgs) == 0 {
		return nil, errors.Errorf("could not find package %s", pkgPath)
	}
	return pkgs[0], nil
}

func FindInterface(name string, files []*ast.File) (*ast.InterfaceType, error) {
	var iface *ast.InterfaceType

	for _, f := range files {
		ast.Inspect(f, func(n ast.Node) bool {
			if t, ok := n.(*ast.TypeSpec); ok {
				if iface != nil {
					return false
				}

				if i, ok := t.Type.(*ast.InterfaceType); ok && t.Name.Name == name {
					iface = i
					return false
				}
			}
			return true
		})

		if iface != nil {
			return iface, nil
		}
	}
	return nil, errors.Errorf("could not find %s interface", name)
}

func FindMethodsCalledOnType(info *types.Info, typ types.Type, caller *ast.FuncDecl) []string {
	var methods []string

	ast.Inspect(caller, func(n ast.Node) bool {
		if s, ok := n.(*ast.SelectorExpr); ok {

			var receiver *ast.Ident
			switch r := s.X.(type) {
			case *ast.Ident:
				// Left-hand side of the selector is an identifier, eg:
				//
				//   a := p.API
				//   a.GetTeams()
				//
				receiver = r
			case *ast.SelectorExpr:
				// Left-hand side of the selector is a selector, eg:
				//
				//   p.API.GetTeams()
				//
				receiver = r.Sel
			}

			if receiver != nil {
				obj := info.ObjectOf(receiver)
				if obj != nil && types.Identical(obj.Type(), typ) {
					methods = append(methods, s.Sel.Name)
				}
				return false
			}

		}
		return true
	})

	return methods
}
