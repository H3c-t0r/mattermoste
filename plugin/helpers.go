// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

package plugin

import (
	"bytes"
	"github.com/mattermost/mattermost-server/model"
	"github.com/mattermost/mattermost-server/services/httpservice"
	"io/ioutil"
	"net/http"
	"net/url"
)

type Helpers interface {
	// EnsureBot either returns an existing bot user matching the given bot, or creates a bot user from the given bot.
	// Returns the id of the resulting bot.
	EnsureBot(bot *model.Bot) (string, error)

	// KVSetJSON stores a key-value pair, unique per plugin, marshalling the given value as a JSON string.
	KVSetJSON(key string, value interface{}) error

	// KVCompareAndSetJSON updates a key-value pair, unique per plugin, but only if the current value matches the given oldValue after marshalling as a JSON string.
	// Inserts a new key if oldValue == nil.
	// Returns (false, err) if DB error occurred
	// Returns (false, nil) if current value != oldValue or key already exists when inserting
	// Returns (true, nil) if current value == oldValue or new key is inserted
	//
	// Minimum server version: 5.12
	KVCompareAndSetJSON(key string, oldValue interface{}, newValue interface{}) (bool, error)

	// KVCompareAndDeleteJSON deletes a key-value pair, unique per plugin, but only if the current value matches the given oldValue after marshalling as a JSON string.
	// Returns (false, err) if DB error occurred
	// Returns (false, nil) if current value != oldValue or the key was already deleted
	// Returns (true, nil) if current value == oldValue
	//
	// Minimum server version: 5.16
	KVCompareAndDeleteJSON(key string, oldValue interface{}) (bool, error)

	// KVGetJSON retrieves a value based on the key, unique per plugin, unmarshalling the previously set JSON string into the given value. Returns true if the key exists.
	KVGetJSON(key string, value interface{}) (bool, error)

	// KVSetWithExpiryJSON stores a key-value pair with an expiry time, unique per plugin, marshalling the given value as a JSON string.
	//
	// Minimum server version: 5.6
	KVSetWithExpiryJSON(key string, value interface{}, expireInSeconds int64) error

	InstallPluginFromUrl(url string, replace bool) (*model.Manifest, *model.AppError)
}
type HelpersImpl struct {
	API         API
	HTTPService httpservice.HTTPService
}

func (p *HelpersImpl) InstallPluginFromUrl(downloadUrl string, replace bool) (*model.Manifest, *model.AppError) {
	parsedUrl, _ := url.Parse(downloadUrl)
	if !*p.API.GetConfig().PluginSettings.AllowInsecureDownloadUrl && parsedUrl.Scheme != "https" {
		return nil, model.NewAppError("InstallPluginFromUrl", "api.plugin.install.insecure_url.app_error", nil, "", http.StatusBadRequest)
	}

	client := p.HTTPService.MakeClient(true)
	response, err := client.Get(downloadUrl)
	if err != nil {
		return nil, model.NewAppError("InstallPluginFromUrlHelper", "api.plugin.install.download_failed.app_error", nil, err.Error(), http.StatusBadRequest)
	}
	defer response.Body.Close()

	fileBytes, err := ioutil.ReadAll(response.Body)
	if err != nil {
		return nil, model.NewAppError("InstallPluginFromUrlHelper", "api.plugin.install.reading_stream_failed.app_error", nil, err.Error(), http.StatusBadRequest)
	}

	manifest, appError := p.API.InstallPlugin(bytes.NewReader(fileBytes), true)
	if appError != nil {
		return nil, appError
	}

	return manifest, nil
}
