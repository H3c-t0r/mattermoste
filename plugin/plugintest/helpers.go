// Code generated by mockery v1.0.0. DO NOT EDIT.

// Regenerate this file using `make plugin-mocks`.

package plugintest

import (
	model "github.com/mattermost/mattermost-server/model"
	plugin "github.com/mattermost/mattermost-server/plugin"
	mock "github.com/stretchr/testify/mock"
)

// Helpers is an autogenerated mock type for the Helpers type
type Helpers struct {
	mock.Mock
}

// EnsureBot provides a mock function with given fields: bot
func (_m *Helpers) EnsureBot(bot *model.Bot) (string, error) {
	ret := _m.Called(bot)

	var r0 string
	if rf, ok := ret.Get(0).(func(*model.Bot) string); ok {
		r0 = rf(bot)
	} else {
		r0 = ret.Get(0).(string)
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(*model.Bot) error); ok {
		r1 = rf(bot)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// ExecuteCommand provides a mock function with given fields: c, args
func (_m *Helpers) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
	ret := _m.Called(c, args)

	var r0 *model.CommandResponse
	if rf, ok := ret.Get(0).(func(*plugin.Context, *model.CommandArgs) *model.CommandResponse); ok {
		r0 = rf(c, args)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*model.CommandResponse)
		}
	}

	var r1 *model.AppError
	if rf, ok := ret.Get(1).(func(*plugin.Context, *model.CommandArgs) *model.AppError); ok {
		r1 = rf(c, args)
	} else {
		if ret.Get(1) != nil {
			r1 = ret.Get(1).(*model.AppError)
		}
	}

	return r0, r1
}

// KVCompareAndSetJSON provides a mock function with given fields: key, oldValue, newValue
func (_m *Helpers) KVCompareAndSetJSON(key string, oldValue interface{}, newValue interface{}) (bool, error) {
	ret := _m.Called(key, oldValue, newValue)

	var r0 bool
	if rf, ok := ret.Get(0).(func(string, interface{}, interface{}) bool); ok {
		r0 = rf(key, oldValue, newValue)
	} else {
		r0 = ret.Get(0).(bool)
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(string, interface{}, interface{}) error); ok {
		r1 = rf(key, oldValue, newValue)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// KVGetJSON provides a mock function with given fields: key, value
func (_m *Helpers) KVGetJSON(key string, value interface{}) (bool, error) {
	ret := _m.Called(key, value)

	var r0 bool
	if rf, ok := ret.Get(0).(func(string, interface{}) bool); ok {
		r0 = rf(key, value)
	} else {
		r0 = ret.Get(0).(bool)
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(string, interface{}) error); ok {
		r1 = rf(key, value)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// KVSetJSON provides a mock function with given fields: key, value
func (_m *Helpers) KVSetJSON(key string, value interface{}) error {
	ret := _m.Called(key, value)

	var r0 error
	if rf, ok := ret.Get(0).(func(string, interface{}) error); ok {
		r0 = rf(key, value)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// KVSetWithExpiryJSON provides a mock function with given fields: key, value, expireInSeconds
func (_m *Helpers) KVSetWithExpiryJSON(key string, value interface{}, expireInSeconds int64) error {
	ret := _m.Called(key, value, expireInSeconds)

	var r0 error
	if rf, ok := ret.Get(0).(func(string, interface{}, int64) error); ok {
		r0 = rf(key, value, expireInSeconds)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// RegisterCommand provides a mock function with given fields: command, callback
func (_m *Helpers) RegisterCommand(command *model.Command, callback plugin.CommandCallback) error {
	ret := _m.Called(command, callback)

	var r0 error
	if rf, ok := ret.Get(0).(func(*model.Command, plugin.CommandCallback) error); ok {
		r0 = rf(command, callback)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}
