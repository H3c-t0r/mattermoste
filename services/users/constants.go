// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

package users

const MissingAccountError = "app.user.missing_account.const"
const MissingAuthAccountError = "app.user.get_by_auth.missing_account.app_error"
