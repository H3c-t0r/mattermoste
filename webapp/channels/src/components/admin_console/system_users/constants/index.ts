// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum ColumnNames {
    username = 'usernameColumn',
    displayName = 'displayNameColumn',
    email = 'emailColumn',
    createAt = 'createAtColumn',
    lastLoginAt = 'lastLoginColumn',
    lastStatusAt = 'lastStatusAtColumn',
    lastPostDate = 'lastPostDateColumn',
    daysActive = 'daysActiveColumn',
    totalPosts = 'totalPostsColumn',
    actions = 'actionsColumn',
}
