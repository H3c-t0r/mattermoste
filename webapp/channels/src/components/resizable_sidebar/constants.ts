// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum SidebarSize {
    SMALL='small',
    MEDIUM= 'medium',
    LARGE= 'large',
    XLARGE= 'xLarge',
}

export enum ResizeDirection {
    LEFT = 'left',
    RIGHT = 'right',
}

export enum CssVarKeyForResizable {
    LHS = 'overrideLhsWidth',
    RHS = 'overrideRhsWidth',
}

export const SIDEBAR_SNAP_SIZE = 16;
export const SIDEBAR_SNAP_SPEED_LIMIT = 5;

export const DEFAULT_LHS_WIDTH = 240;

export const LHS_MIN_MAX_WIDTH: { [size in SidebarSize]: { min: number; max: number}} = {
    [SidebarSize.SMALL]: {
        min: 240,
        max: 240,
    },
    [SidebarSize.MEDIUM]: {
        min: 200,
        max: 264,
    },
    [SidebarSize.LARGE]: {
        min: 200,
        max: 304,
    },
    [SidebarSize.XLARGE]: {
        min: 200,
        max: 440,
    },
};

export const RHS_MIN_MAX_WIDTH: { [size in SidebarSize]: { min: number; max: number; default: number}} = {
    [SidebarSize.SMALL]: {
        min: 400,
        max: 400,
        default: 400,
    },
    [SidebarSize.MEDIUM]: {
        min: 304,
        max: 400,
        default: 400,
    },
    [SidebarSize.LARGE]: {
        min: 304,
        max: 464,
        default: 400,
    },
    [SidebarSize.XLARGE]: {
        min: 304,
        max: 776,
        default: 500,
    },
};
