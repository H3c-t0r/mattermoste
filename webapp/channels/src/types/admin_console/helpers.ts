export type ConsoleAccess = {
    read: Record<string, boolean>;
    write: Record<string, boolean>;
};

/**
 * Flattens the properties of an object recursively.
 *
 * @template T - The input type to flatten.
 * @param {T} obj - The object to be flattened.
 * @returns {T extends object ? { [K in keyof T]: T[K] } : never} - The flattened object.
 *
 * @example
 * // Flattening an object
 * const obj = {
 *   prop1: {
 *     nestedProp1: 1,
 *     nestedProp2: 'two',
 *   },
 *   prop2: {
 *     nestedProp3: true,
 *     nestedProp4: ['a', 'b', 'c'],
 *   },
 * };
 *
 * type FlattenedObj = FlattenProperties<typeof obj>;
 * // FlattenedObj is equivalent to:
 * // {
 * //   prop1: {
 * //     nestedProp1: 1,
 * //     nestedProp2: 'two',
 * //   },
 * //   nestedProp1: 1,
 * //   nestedProp2: 'two',
 * //   prop2: {
 * //     nestedProp3: true,
 * //     nestedProp4: ['a', 'b', 'c'],
 * //   },
 * //   nestedProp3: true,
 * //   nestedProp4: ['a', 'b', 'c'],
 * // }
 */
export type FlattenProperties<T> = T extends object
    ? { [K in keyof T]: T[K] }
    : never;
