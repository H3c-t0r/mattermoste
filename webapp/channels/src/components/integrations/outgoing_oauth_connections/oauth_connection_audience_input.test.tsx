// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';

import type {OutgoingOAuthConnection} from '@mattermost/types/integrations';

import {Permissions} from 'mattermost-redux/constants';

import OAuthConnectionAudienceInput from 'components/integrations/outgoing_oauth_connections/oauth_connection_audience_input';

import {mountWithIntl} from 'tests/helpers/intl-test-helper';
import mockStore from 'tests/test_store';

describe('components/integrations/outgoing_oauth_connections/OAuthConnectionAudienceInput', () => {
    const connection: OutgoingOAuthConnection = {
        id: 'facxd9wpzpbpfp8pad78xj75pr',
        name: 'testConnection',
        client_secret: '',
        client_id: 'clientid',
        create_at: 1501365458934,
        creator_id: '88oybd1dwfdoxpkpw1h5kpbyco',
        update_at: 1501365458934,
        grant_type: 'client_credentials',
        oauth_token_url: 'https://token.com',
        audiences: ['https://aud.com'],
    };

    const outgoingOAuthConnections: Record<string, OutgoingOAuthConnection> = {
        facxd9wpzpbpfp8pad78xj75pr: connection,
    };

    const baseProps: React.ComponentProps<typeof OAuthConnectionAudienceInput> = {
        value: '',
        onChange: jest.fn(),
        placeholder: '',
    };

    const state = {
        entities: {
            general: {
                config: {
                    EnableOutgoingOAuthConnections: 'true',
                },
                license: {
                    IsLicensed: 'true',
                    Cloud: 'true',
                },
            },
            users: {
                currentUserId: 'current_user_id',
                profiles: {
                    current_user_id: {roles: 'system_role'},
                },
            },
            roles: {
                roles: {
                    system_role: {id: 'system_role', permissions: [Permissions.MANAGE_OUTGOING_OAUTH_CONNECTIONS]},
                },
            },
            integrations: {
                outgoingOAuthConnections,
            },
        },
    };

    test('should match snapshot with no existing connections', () => {
        const props = {...baseProps};
        const newState = {...state, integrations: {outgoingOAuthConnections: {}}};
        const store = mockStore(newState);
        const wrapper = mountWithIntl(
            <Router>
                <Provider store={store}>
                    <OAuthConnectionAudienceInput {...props}/>
                </Provider>
            </Router>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with existing connections', () => {
        const props = {...baseProps};
        const store = mockStore(state);
        const wrapper = mountWithIntl(
            <Router>
                <Provider store={store}>
                    <OAuthConnectionAudienceInput {...props}/>
                </Provider>
            </Router>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot when typed in value matches a configured audience', () => {
        const props = {...baseProps, value: 'https://aud.com'};
        const store = mockStore(state);
        const wrapper = mountWithIntl(
            <Router>
                <Provider store={store}>
                    <OAuthConnectionAudienceInput {...props}/>
                </Provider>
            </Router>,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
