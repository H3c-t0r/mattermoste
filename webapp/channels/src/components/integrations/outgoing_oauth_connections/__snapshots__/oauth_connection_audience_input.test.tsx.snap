// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/integrations/outgoing_oauth_connections/OAuthConnectionAudienceInput should match snapshot when typed in value matches a configured audience 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "undefinedundefined",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <OAuthConnectionAudienceInput
        onChange={[MockFunction]}
        placeholder=""
        value="https://aud.com"
      >
        <input
          autoComplete="off"
          className="form-control"
          id="url"
          maxLength={1024}
          onChange={[MockFunction]}
          placeholder=""
          value="https://aud.com"
        />
        <div
          className="outgoing-oauth-audience-match-message"
        >
          <_StyledOauthIcon
            size={20}
          >
            <OauthIcon
              className="_StyledOauthIcon-eWpFZy hGBbfE"
              size={20}
            >
              <svg
                className="_StyledOauthIcon-eWpFZy hGBbfE"
                fill="currentColor"
                height={20}
                version="1.1"
                viewBox="0 0 24 24"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12,20c-4.4,0-8-3.6-8-8s3.6-8,8-8s8,3.6,8,8S16.4,20,12,20 M12,2C6.5,2,2,6.5,2,12s4.5,10,10,10s10-4.5,10-10S17.5,2,12,2z M12,5c-3.9,0-7,3.1-7,7s3.1,7,7,7s7-3.1,7-7S15.9,5,12,5z M14.8,16c-0.1,0-0.2,0-0.3,0c-0.4,0-0.8-0.3-0.9-0.6L13.1,14h-2.3
	l-0.4,1.4C10.3,15.7,9.9,16,9.5,16c-0.1,0-0.2,0-0.3,0c-0.5-0.1-0.8-0.7-0.6-1.2l1.9-6.1C10.7,8.3,11,8,11.4,8h1c0.1,0,0.2,0,0.3,0
	c0.2,0.1,0.4,0.2,0.5,0.3c0.1,0.1,0.1,0.2,0.1,0.3l2,6.1C15.5,15.3,15.3,15.8,14.8,16z M7,12c0-1.7,0.9-3.3,2.3-4.2
	C9.2,7.9,9.2,8.1,9.1,8.2l-1.8,5.6C7.1,13.2,7,12.6,7,12z M12,17c-0.3,0-0.6,0-0.9-0.1c0.3-0.3,0.6-0.7,0.7-1.1l0.1-0.3H12l0.1,0.4
	c0.1,0.4,0.4,0.8,0.7,1.1C12.6,17,12.3,17,12,17z M16.6,13.8l-1.9-5.6c0-0.2-0.1-0.3-0.2-0.4C16,8.6,17,10.2,17,12
	C17,12.6,16.9,13.2,16.6,13.8z"
                />
              </svg>
            </OauthIcon>
          </_StyledOauthIcon>
          <strong
            style={
              Object {
                "left": "42px",
                "position": "absolute",
                "top": "45px",
              }
            }
          >
            <FormattedMessage
              defaultMessage="Connected to \\"{connectionName}\\""
              id="add_outgoing_oauth_connection.connected"
              values={
                Object {
                  "connectionName": "testConnection",
                }
              }
            >
              <span>
                Connected to "testConnection"
              </span>
            </FormattedMessage>
          </strong>
        </div>
      </OAuthConnectionAudienceInput>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`components/integrations/outgoing_oauth_connections/OAuthConnectionAudienceInput should match snapshot with existing connections 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "undefinedundefined",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <OAuthConnectionAudienceInput
        onChange={[MockFunction]}
        placeholder=""
        value=""
      >
        <input
          autoComplete="off"
          className="form-control"
          id="url"
          maxLength={1024}
          onChange={[MockFunction]}
          placeholder=""
          value=""
        />
        <div
          className="outgoing-oauth-audience-match-message"
        >
          <_StyledInformationOutlineIcon
            size={20}
          >
            <InformationOutlineIcon
              className="_StyledInformationOutlineIcon-fJlbfB cjrhWO"
              size={20}
            >
              <svg
                className="_StyledInformationOutlineIcon-fJlbfB cjrhWO"
                fill="currentColor"
                height={20}
                version="1.1"
                viewBox="0 0 24 24"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
                />
              </svg>
            </InformationOutlineIcon>
          </_StyledInformationOutlineIcon>
          <strong
            style={
              Object {
                "left": "42px",
                "position": "absolute",
                "top": "45px",
              }
            }
          >
            <FormattedMessage
              defaultMessage="Not linked to an OAuth connection"
              id="add_outgoing_oauth_connection.not_connected"
            >
              <span>
                Not linked to an OAuth connection
              </span>
            </FormattedMessage>
          </strong>
        </div>
      </OAuthConnectionAudienceInput>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`components/integrations/outgoing_oauth_connections/OAuthConnectionAudienceInput should match snapshot with no existing connections 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "undefinedundefined",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <OAuthConnectionAudienceInput
        onChange={[MockFunction]}
        placeholder=""
        value=""
      >
        <input
          autoComplete="off"
          className="form-control"
          id="url"
          maxLength={1024}
          onChange={[MockFunction]}
          placeholder=""
          value=""
        />
        <div
          className="outgoing-oauth-audience-match-message"
        >
          <_StyledInformationOutlineIcon
            size={20}
          >
            <InformationOutlineIcon
              className="_StyledInformationOutlineIcon-fJlbfB cjrhWO"
              size={20}
            >
              <svg
                className="_StyledInformationOutlineIcon-fJlbfB cjrhWO"
                fill="currentColor"
                height={20}
                version="1.1"
                viewBox="0 0 24 24"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20,12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M11,17H13V11H11V17Z"
                />
              </svg>
            </InformationOutlineIcon>
          </_StyledInformationOutlineIcon>
          <strong
            style={
              Object {
                "left": "42px",
                "position": "absolute",
                "top": "45px",
              }
            }
          >
            <FormattedMessage
              defaultMessage="Not linked to an OAuth connection"
              id="add_outgoing_oauth_connection.not_connected"
            >
              <span>
                Not linked to an OAuth connection
              </span>
            </FormattedMessage>
          </strong>
        </div>
      </OAuthConnectionAudienceInput>
    </Provider>
  </Router>
</BrowserRouter>
`;
